core/templates/save-lazy-all.tid

```diff
@@ -1,9 +1,9 @@
 title: $:/core/save/lazy-all
 
 \define saveTiddlerFilter()
-[is[system]] -[prefix[$:/state/popup/]] -[[$:/HistoryList]] -[[$:/boot/boot.css]] -[type[application/javascript]library[yes]] -[[$:/boot/boot.js]] -[[$:/boot/bootprefix.js]] +[sort[title]] 
+[is[system]] -[prefix[$:/state/popup/]] -[[$:/HistoryList]] -[[$:/boot/boot.css]] -[type[application/javascript]] -[[$:/boot/boot.js]] -[[$:/boot/bootprefix.js]] +[sort[title]] 
```

Comment:
The construction `-[type[application/javascript]library[yes]]` is used in the core as a rather clumsy way to exclude the SJCL library when saving. The same construction is in the usual `$:/core/save/all` filter too.

It's possible that we should review unintended side effects of that behaviour, but here we should leave it alone. I think the correct value for this filter should be:

```
[is[system]] -[prefix[$:/state/popup/]] -[[$:/HistoryList]] -[[$:/boot/boot.css]] -[type[application/javascript]library[yes]] -[[$:/boot/boot.js]] -[[$:/boot/bootprefix.js]] +[is[tiddler]type[application/javascript]] +[sort[title]]
```

And the skinny filter should be as you have it:

```
[!is[system]] -[type[application/javascript]]
```

Reply:
Updated and tested locally.

And an interesting thing is, even you may misuse `+[[]]` to try to add a set to filter result, while the correct syntax is just `[[]]`, lol.

I always think using `+` as "pipe" operator is not a very good idea...maybe `=` is better, but at this point, we might not have a chance to change this.

Reply:
> And an interesting thing is, even you may misuse +[[]] to try to add a set to filter result, while the correct syntax is just [[]], lol.

You're quite right, that's a typo, sorry!

> I always think using + as "pipe" operator is not a very good idea...maybe = is better, but at this point, we might not have a chance to change this.

Yes, I'm not sure that + was the best choice.